### PR TITLE
LinkTag: correctly display organization avatar

### LINKED ISSUE #11880: Liquid Tags does not respect company avatar
**Describe the bug**

When using liquid tags on a post that was published under an organization it shows the users avatar.

**To Reproduce**

1. Create a post under an organization
2. Add it as liquid tag into a comment or another post
3. The avatar displayed will be the poster's.

**Expected behavior**

Everywhere the way a organization post's avatar is the organization's avatar and in small the poster's avatar (notification, dashboard, homepage).

When using the liquid tags it should show the avatar in the same way

**Screenshots**

Organizaiton's post avatar:

Screenshot 2020-12-14 at 10 02 49 AM

On the homepage the avatar is the same style:

Screenshot 2020-12-14 at 10 03 57 AM

When using Liquid tags we don't get this behavior:

Screenshot 2020-12-14 at 10 05 01 AM

**Desktop (please complete the following information):**

- OS, version: MacOS
- Browser, version: Firefox 83.0

### PR DESCRIPTION
**Pull request by meg-gutshall**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This was part of the Live Code Pairing series with @nickytonline and @cmgorton. For this stream we worked on fixing the way avatars appear in liquid tag links when a post is written by a user, but under an organization account. The avatar was not rendering as displayed on the front page, it was only showing the user's avatar.

## Related Tickets & Documents

Fixes #11880 

## QA Instructions, Screenshots, Recordings

This is what we were looking for on the Liquid tags:
Screen Shot 2021-06-07 at 2 48 58 AM

This is the issue that we were seeing:
Screen Shot 2021-06-06 at 10 39 26 PM

Here is the fix:
Screen Shot 2021-06-07 at 2 48 35 AM

Here's an alternate with the author pointing solely to the organization:
Screen Shot 2021-06-07 at 2 46 58 AM

### UI accessibility concerns?

The images still retain `alt` tags and they're both wrapped in the hyperlink that will redirect to the article's author.

## Added tests?

- [ ] Yes
- [x] No, and this is why: The existing tests would need a decent amount of work to get to the granularity needed to actually test if this feature was working and I don't have that kind of time right now.
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Review by jgaskins — COMMENTED** [review]
Hey Meg, thanks for the PR! :100: It looks great! Just a couple nitpicks — and one of them is just a suggestion that I'm not even sure is feasible. :joy:

Admittedly I haven't watched the full stream so I don't know if y'all talked about these changes or not. Either way, it was great to see a Ruby for Good friend come through my GitHub notifications. 🙌🏼

File: db/schema.rb [hunk 1]
```diff
@@ -1423,9 +1423,9 @@
 
   create_table "users_gdpr_delete_requests", force: :cascade do |t|
     t.datetime "created_at", precision: 6, null: false
-    t.string "email"
+    t.string "email", null: false
     t.datetime "updated_at", precision: 6, null: false
-    t.integer "user_id"
+    t.integer "user_id", null: false
```
[the remark this answers is not in the document]
Comment (meg-gutshall) [reply db/schema.rb:1428]: Sure thing! I think that was automatically added when I ran the migrations. (I couldn't get Docker to cooperate with me so I ended up building locally instead.)

File: app/views/articles/_liquid.html.erb [hunk 2]
```diff
@@ -7,22 +7,44 @@
...
+      <div class='ltag__link__content'>
+        <h2><%= title %></h2>
+        <h3><%= article.user.name %> for <%= article.organization.name %> ・ <%= article.readable_publish_date %> ・ <%= article.reading_time < 2 ? 1 : article.reading_time %> min read</h3>
+        <div class='ltag__link__taglist'>
+        <% article.tag_list.each do |t| %>
+          <span class='ltag__link__tag'>#<%= t %></span>
+        <% end %>
+        </div>
+      </div>
+    </a>
```
[the remark this answers is not in the document]
Comment (meg-gutshall) [reply app/views/articles/_liquid.html.erb:29]: Okay, I'll give this another go and see what I can do. I was fading when I got this in last night so I can probably refactor a bit in here.

**Comment by meg-gutshall** [comment]
Oh hi there @jgaskins! Yes, so lovely to see another gooder! 😁 



BTW, when I tried to run the RSpec tests last night before submitting, I saw that the test db hadn't been created in the initial setup. Is there a different command I need to run for that?

Thank you for the helpful feedback! I will try to make time this weekend to submit an update.

File: app/views/articles/_liquid.html.erb [hunk 3]
```diff
@@ -7,22 +7,44 @@
...
+  <% else %>
+    <a href='<%= article.user.path %>' class='ltag__link__link'>
+      <div class='ltag__link__pic'>
+        <img src='<%= Images::Profile.call(article.user.profile_image_url, length: 150) %>' alt='<%= "#{article.user.username} image" %>'>
```
Comment (aitchiss) [inline app/views/articles/_liquid.html.erb:33]: nitpick (blocking) - like the example above, removing the word "image" helps reduce unnecessary noise for screen reader users. I know it was like this before, but it's a great opportunity to sweep it up now 🙌 

```suggestion
        <img src='<%= Images::Profile.call(article.user.profile_image_url, length: 150) %>' alt='<%= "#{article.user.username}" %>'>
```

**Review by aitchiss — CHANGES_REQUESTED** [review]
I tested this locally and it's working great!

I made sure there's no regression for links in posts published before the changes, as well as for newly added ones, and all looks good to me 👍 

I just have a couple of very minor change requests to enhance the experience for screen reader users. Let me know if you have any questions, and thanks very much for these changes!

**Comment by meg-gutshall** [comment]
@aitchiss I appreciate the work you put into the testing portion of this! I agree that alt tag content sounds much better. I've accepted both of your suggested changes.

@citizen428 I did run `bin/setup` per the instructions but somehow the test database did not get created, only the development database. Here is the content of my `test.log` file if that helps any:

```
Unset ENV variable: IMGPROXY_KEY.
Unset ENV variable: IMGPROXY_SALT.
Unset ENV variable: RELEASE_FOOTPRINT.
** [Honeybadger] Initializing Honeybadger Error Tracker for Ruby. Ship it! version=4.8.0 framework=rails level=1 pid=34131
Unset ENV variable: COMMUNITY_COPYRIGHT_START_YEAR.
```

**Comment by aitchiss** [comment]
@meg-gutshall - this is looking great - thanks for making those tweaks!

I noticed there are some specs failing
- The "frontend tests" build job failed due to a known flakey test - I don't think it's anything to do with your code and I've restarted it now 🙂 
- The other failed build job is because of the changes to the `alt` tag - in `spec/liquid_tags/link_tag_spec.rb` we'll need to update the `correct_link_html` to reflect the new alt text. Let me know if you have any trouble doing this - hopefully just a quick change!

Thanks again - once the specs are passing I think this is good to go 🚀

**Comment by citizen428** [comment]
Hi @meg-gutshall thanks for all your work on this PR so far, it's much appreciated! 🥇 Is there anything we can do to help you push this over the finish line?

**Comment by meg-gutshall** [comment]
Hi @aitchiss and @citizen428! Thank you for your help and patience. I'm finishing up a coding bootcamp with a deadline of 6/30 (and *just starting* my final project today 😬 ) so I need to prioritize that.

I can finish up this PR in early July if that's okay with you. If it can't wait, I'm okay with co-authoring this PR. I feel like @aitchiss should get co-author recognition if she hasn't already for her suggested changes.

Please let me know how you would like to proceed!

**Comment by citizen428** [comment]
I'm fine either way, I'll leave it up to @aitchiss 😃

**Comment by aitchiss** [comment]
That's exciting about your final project @meg-gutshall!

We're not in any rush, but I think there's only a teeny change to the specs needed, so I'm happy to have a look at that and then hopefully we can get this merged 😄  You've already done 99.9% - happy to help push it over the line!

**Comment by meg-gutshall** [comment]
Woohoo! Thank you tons! Time for a celebration dance!!!

**Comment by cmgorton** [comment]
> Woohoo! Thank you tons! Time for a celebration dance!!!


Woohoo! Thanks for the contribution Meg🎉

And thanks team for all the help 🙏

**Comment by nickytonline** [comment]
Nice to see this completed @meg-gutshall! I'm going to link the stream to this PR for folks to see it come full circle.

Thanks again for your contribution to the Forem codebase! 😎

<p align="center">
	<a href="https://www.youtube.com/watch?v=lpdzrcx60Ho&list=PL8Bb2WLXhUaBGrYRnmDaF73LDCjEGzjIE&index=2"></a>
</p>

File: db/schema.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (citizen428) [reply db/schema.rb]: @meg-gutshall I fixed this for you so we can get this PR merged 😃

**Review by citizen428 — APPROVED** [review]

**Pull request merged**